src/q_learn.jl

```diff
@@ -25,26 +25,27 @@ Parameters:
     print information during training
     default: `true`
 """
-@with_kw mutable struct QLearningSolver{E<:ExplorationPolicy} <: Solver
+Base.@kwdef mutable struct QLearningSolver{E<:ExplorationPolicy,RNG<:AbstractRNG} <: Solver
```

Comment:
Random question...why does this need to be mutable?

Reply:
I suppose it doesn't *need* to be mutable. However, the mutability doesn't seem to have a big impact on performance so I just left it as is.